wordpress-mobile/WordPress-Android
# Introduce empty view handling to the improved My Site fragment

Fixes #13582

This PR introduces the ability for the improved My Site fragment to show the following view when no site is selected:

<img width=300 src="https://user-images.githubusercontent.com/830056/102273616-5cd65680-3f01-11eb-845e-c33bd1fa5214.jpeg"/>

### To test

Make sure all unit tests in `MySiteViewModelTest` pass. 

Additionally:

1. Turn on the Improved My Site flag and restart the app.
1. Login with an account that has no sites.
1. Go to the My Site screen.
1. Notice the empty view.
1. Click **Add New Site**.
1. Notice the _Add new site dialog_.
1. Choose **Add self-hosted site**.
1. Follow the flow and make sure the newly added site appears in the My Site screen in the end.
1. Remove the self-hosted site.
1. Notice the empty view again.
1. Click **Add New Site**.
1. Notice the _Add new site dialog_.
1. Choose **Create WordPress.com site**.
1. Follow the flow and make sure the newly created site appears in the My Site screen in the end.

----

PR submission checklist:
